src/webhooks/kafka-control-plane/kafka-control-plane.ts

```diff
@@ -0,0 +1,108 @@
+import * as Sentry from '@sentry/node';
+import { KnownBlock } from '@slack/types';
+import { FastifyReply, FastifyRequest } from 'fastify';
+
+import { KafkaControlPlaneResponse } from '@types';
+
+import { bolt } from '@/api/slack';
+import * as slackblocks from '@/blocks/slackBlocks';
+import {
+  KAFKA_CONTROL_PLANE_CHANNEL_ID,
+  KAFKA_CONTROL_PLANE_WEBHOOK_SECRET,
+} from '@/config';
+import { verifySignature } from '@/utils/verifySignature';
+
+export async function handler(
+  request: FastifyRequest<{ Body: KafkaControlPlaneResponse }>,
+  reply: FastifyReply
+) {
+  try {
+    const clientSignatureHeader =
+      request.headers['x-infra-event-notifier-signature'] ?? '';
+    const clientSignature = Array.isArray(clientSignatureHeader)
+      ? clientSignatureHeader.join('')
+      : clientSignatureHeader;
+
+    const payloadBody = request.body ? JSON.stringify(request.body) : '';
+    const isVerified = verifySignature(
+      payloadBody,
+      clientSignature!,
+      KAFKA_CONTROL_PLANE_WEBHOOK_SECRET!,
+      (i) => i,
+      'sha256'
+    );
+
+    if (!isVerified) {
+      return reply.code(401).send('Unauthorized');
+    }
```

Comment:
I'd recommend utilizing [extractAndVerifySignature](https://github.com/getsentry/eng-pipes/blob/fb8c72ff8d4939f5736388810b4ba9124d397195/src/utils/extractAndVerifySignature.ts#L7) to reduce code duplication similar to how it is used in the gocd webhook https://github.com/getsentry/eng-pipes/blob/fb8c72ff8d4939f5736388810b4ba9124d397195/src/webhooks/gocd/gocd.ts#L19-L29.